### PR TITLE
fix DefInfo not loading

### PR DESCRIPTION
**Pull request by neelance**
For some reason the backend is returning inconsistent data, listing a file as an example but then returning no reference in that file. This change makes the UI not fail in that case.

Fixes https://app.asana.com/0/87040567695724/156775931921647

### CODE REVIEW

File: app/web_modules/sourcegraph/def/RefsContainer.js [hunk 1]
```diff
@@ -184,7 +184,7 @@ export default class RefsContainer extends Container {
 			Dispatcher.Backends.dispatch(new DefActions.WantDef(repo, rev, def));
 		}
 
-		if (nextState.refs && !nextState.refs.Error && (nextState.refs !== prevState.refs || nextState.shownFiles !== prevState.shownFiles)) {
+		if (nextState.refs && nextState.refs.length > 0 && !nextState.refs.Error && (nextState.refs !== prevState.refs || nextState.shownFiles !== prevState.shownFiles)) {
```
Comment (rothfels) [inline app/web_modules/sourcegraph/def/RefsContainer.js]: Do we prefer the style `&& nextState.refs.length > 0 &&` or the `&& nextState.refs.length &&`?

File: app/web_modules/sourcegraph/def/DefBackend.js [hunk 2]
```diff
@@ -118,8 +118,10 @@ const DefBackend = {
 							.then(checkStatus)
 							.then((resp) => resp.json())
 							.catch((err) => ({Error: err}))
-							.then(convNotFound)
 							.then((data) => {
+								if (get(data, "Error.response.status") === 404) {
+									data = [];
```
Comment (rothfels) [inline app/web_modules/sourcegraph/def/DefBackend.js]: :( bummer

**Comment by rothfels** [comment]
Sad to introduce more debt, but fine to merge.

File: app/web_modules/sourcegraph/def/RefsContainer.js (hunk 1 above)
Comment (neelance) [inline app/web_modules/sourcegraph/def/RefsContainer.js]: I personally prefer to make explicit that it is an integer, not a boolean. Implicit conversions are a source of pain. ;-)

**Pull request closed without merging**